lms/djangoapps/django_comment_client/forum/tests.py

```diff
@@ -531,9 +531,133 @@ def test_group_info_in_ajax_response(self, mock_request):
         )
 
 
+@patch('requests.request')
+class SingleThreadContentGroupTestCase(ContentGroupTestCase):
+    def assert_can_access(self, user, discussion_id, thread_id, should_have_access):
+        """
+        Verify that a user has access to a thread within a given
+        discussion_id when should_have_access is True, otherwise
+        verify that the user does not have access to that thread.
+        """
+        request = RequestFactory().get("dummy_url")
+        request.user = user
+        mako_middleware_process_request(request)
+
+        def call_single_thread():
+            return views.single_thread(
+                request,
+                unicode(self.course.id),
+                discussion_id,
+                thread_id
+            )
+
+        if should_have_access:
+            self.assertEqual(call_single_thread().status_code, 200)
+        else:
+            with self.assertRaises(Http404):
+                call_single_thread()
+
+    def assert_searched_with_discussion_ids(self, mock_request, expected_commentable_ids):
+        """
+        Verify that the comments service was searched for threads with
+        the expected discussion ids (passed to the comments service as
+        'commentable_ids').
+        """
+        mock_request.assert_called_with(
+            'get',
+            StringEndsWithMatcher('threads'),
+            headers=ANY,
+            timeout=ANY,
+            data=None,
+            params=PartialDictMatcher({
```

Comment:
We are verifying that all the key/value pairs in the dictionary passed to `PartialDictMatcher` exist exactly in the params passed to `mock_request`. Additional key/value pairs exist in the params passed to `mock_request` (for example 'user_id'), but I figured those aren't relevant to what we're testing. I'm happy to add them if you still think they should be in there, though.

```python
class PartialDictMatcher(object):
    def __init__(self, expected_values):
        self.expected_values = expected_values

    def __eq__(self, other):
        return all([
            key in other and other[key] == value
            for key, value in self.expected_values.iteritems()
        ])
```